include/dsn/dist/meta_state_service.h

```diff
@@ -25,20 +25,46 @@
  */
 # pragma once
 
-# include <dsn/service_api_cpp.h>
+#include <dsn/service_api_cpp.h>
+#include <string>
+#include <functional>
+#include <dsn/internal/task.h>
 
 namespace dsn
 {
     namespace dist
     {
-        // interface
         class meta_state_service
         {
         public:
-            // void create_node(....callback..)
-            // void put(....callback..);
-            // void get(....callback..);
-            // void destroy_node(....callback..);
+            typedef std::function<void (int ec, std::string&& ret_str)> err_string_callback;
+            typedef std::function<void (int ec, std::vector<std::string>&& ret_strv)> err_stringv_callback;
+            typedef std::function<void (int ec, std::string&& ret_strv, int ret)> err_data_callback;
+            typedef std::function<void (int ec, int ret)> err_state_callback;
+            typedef std::function<void (int ec)> err_callback;
+
+        public:
+            //create a dir
+            virtual void amkdir(const std::string& node,
```

Comment:
Well, all the async functions in zookeeper begin with letter a, and I found it not hard to understand so followed the rule.